improver_tests/calibration/rainforests_calibration/test_ApplyRainForestsCalibrationLightGBM.py

```diff
@@ -208,7 +208,37 @@ def test__prepare_features_dataframe(ensemble_features):
     with pytest.raises(ValueError):
         ApplyRainForestsCalibrationLightGBM(
             model_config_dict={}
-        )._prepare_features_dataframe(ensemble_features)
+        )._prepare_features_array(ensemble_features)
+
+
+def test__evaluate_probabilities(
+    ensemble_features, dummy_lightgbm_models, error_threshold_cube
+):
+    """Test that _evaluate_probabilities populates error_threshold_cube.data with
+    probability data."""
+    plugin = ApplyRainForestsCalibrationLightGBM(model_config_dict={})
+    plugin.tree_models, plugin.error_thresholds = dummy_lightgbm_models
+    input_dataset, feature_variables = plugin._prepare_features_array(ensemble_features)
+    precipitation_ind = feature_variables.index("lwe_thickness_of_precipitation_amount")
+    precip_forecast = input_dataset[:, precipitation_ind]
+    data_before = error_threshold_cube.data.copy()
+    plugin._evaluate_probabilities(
+        precip_forecast, input_dataset, error_threshold_cube.data
+    )
+    diff = error_threshold_cube.data - data_before
+    # check each error threshold has been populated
+    assert np.all(np.any(diff != 0, axis=0))
```

Comment:
Would it not suffice to simply check that there are non-zero values in error_threshold_cube?

Reply:
I don't think so, because even before applying `_evaluate_probabilities`, `error_threshold_cube` is already populated with probability data. So the idea of this code is to check that the data is not the same as previously (and has been updated for each threshold).

Reply:
OK, that's a good point. However, I'm still not entirely sure what the intention of this condition is? Is it simply to check that there are difference between the arrays along the threshold dimension?

Reply:
The idea was just to check that the code didn't just update, e.g. the first threshold and leave the others unchanged.